cmd/apiserver-watcher/run.go

```diff
@@ -139,45 +139,51 @@ func runRunCmd(cmd *cobra.Command, args []string) error {
 }
 
 func newHandler(uri *url.URL) (*handler, error) {
-	h := handler{}
+	h := handler{
+		vips: []string{},
+	}
 
 	if runOpts.vip != "" {
-		h.vip = runOpts.vip
+		h.vips[0] = runOpts.vip
```

Comment:
this won't work; `h.vips` is 0-length so that's an overflow

fwiw the `--vip` flag seems to be unused so you could just drop it

Reply:
puff, :face_with_head_bandage: embarrasing XD

Reply:
it seems unused indeed, but I was afraid it was needed